incubator/hnc/pkg/testutils/testutils.go

```diff
@@ -257,13 +263,34 @@ func RecoverHNC() {
 	}
 	// give HNC enough time to repair
 	time.Sleep(5 * time.Second)
-	// Verify and wait till HNC is fully repaired, sometimes it takes up to 30s.
-	// The `kubectl hns create` command will fail is HNC is broken, so we confirm that HNC is back by successfully 
-	// running this command. 
-	CleanupNamespaces("a", "b")
-	MustRun("kubectl create ns a")
-	RunShouldContain("Successfully created", 30, "kubectl hns create b -n a")
-	CleanupNamespaces("a", "b")
+	// Verify and wait till HNC is fully repaired, sometimes it takes up to 30s. We try to create a
+	// subnamespace and wait for it to be created to show that both the validators and reconcilers are
+	// up and running.
+	const (
+		a = "recover-test-a"
+		b = "recover-test-b"
+	)
+	// Do NOT use CleanupNamespaces because that just assumes that if it can't delete a namespace that
+	// everthing's fine, but this is a poor assumption if HNC has just been repaired.
+	//
+	// TODO: if CleanupNamespaces ever starts using labels to select namespaces to delete, then get
+	// rid of this hack.
+	if err := TryRunQuietly("kubectl get ns", a); err == nil {
+		MustRunWithTimeout(30, "kubectl hns set", a, "-a")
+		MustRunWithTimeout(30, "kubectl delete ns", a)
+	}
+	if err := TryRunQuietly("kubectl get ns", b); err == nil {
+		MustRunWithTimeout(30, "kubectl annotate ns", b, "hnc.x-k8s.io/subnamespaceOf-")
+		MustRunWithTimeout(30, "kubectl delete ns", b)
+	}
```

Comment:
Hmm, this looked the same to me as `CleanupNamespaces()` except skipping de-annotating a and no setting AC for b. What makes a difference?

Reply:
CleanupNamespaces is TryRunQuietly, this is MustRunWithTimeout.